### PR TITLE
Moved funds sweep SPV proofs

### PR DESCRIPTION
**Pull request by tomaszslabon**
#Refs: https://github.com/keep-network/keep-core/issues/3738.
This PR adds functionalities for creating and submitting moved funds sweep transaction SPV proofs.

Finding the possible unproven moved funds sweep transactions is done by retrieving `MovingFundsCommitmentSubmitted` events and looking at the recent transactions of target wallets from each event.

An unproven moved funds seep transaction must have the following format:
- the first input must refer to an unproven moved funds sweep request
- the second input (optional) must refer to the current wallet's main UTXO  
- the (single) output that pays to the wallet itself

Once the unproven moved funds sweep transactions are found, they are proven.

### CODE REVIEW

File: pkg/maintainer/spv/moved_funds_sweep.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+			if isUnproven {
+				unprovenMovedFundsSweepTransactions = append(
+					unprovenMovedFundsSweepTransactions,
+					transaction,
+				)
+
+				// A wallet can have only one unproven moved funds sweep
+				// transaction at a time. If we found such transaction, we don't
+				// have to look at this wallet's transactions anymore.
+				break
```
Comment (lukasz-zimnoch) [inline pkg/maintainer/spv/moved_funds_sweep.go:239]: Just checking my understanding: Here we assume that the given wallet has always 0 or 1 unproven transaction (of any type) and `isUnprovenMovedFundsSweepTransaction` filters out previous moved funds sweep transactions that are already proven (using the `movedFundsSweepRequest.State != tbtc.MovedFundsStatePending` check). Is that correct?

File: pkg/maintainer/spv/moving_funds_test.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func TestSubmitMovingFundsProof(t *testing.T) {
 	spvChain := newLocalChain()
 
 	// Take an arbitrary moving funds transaction:
-	// https://live.blockcypher.com/btc-testnet/tx/e6218018ed1874e73b78e16a8cf4f5016cbc666a3f9179557a84083e3e66ff7c
+	// https://live.blockcypher.com/btc-testnet/tx/e6218018ed1874e73b78e16a8cf4f5016cbc666a3f9179557a84083e3e66ff7c/
```
Comment (lukasz-zimnoch) [inline pkg/maintainer/spv/moving_funds_test.go:41]: Looks like an accidental change.

File: pkg/maintainer/spv/moved_funds_sweep.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+	}
+
+	// Some target wallets may appear on the list multiple times. It can happen
+	// if multiple source wallets used the same target wallet. Make a list
+	// of unique wallets.
+	walletPublicKeyHashes := uniqueKeyHashes(targetWallets)
+
+	unprovenMovedFundsSweepTransactions := []*bitcoin.Transaction{}
+
+	for _, walletPublicKeyHash := range walletPublicKeyHashes {
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply pkg/maintainer/spv/moved_funds_sweep.go]: Done in cbd5bf87d03e051b5d8720a7d51e76622959921d.

File: pkg/maintainer/spv/moved_funds_sweep.go [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+		if wallet.State != tbtc.StateLive &&
+			wallet.State != tbtc.StateMovingFunds {
+			// The wallet can only submit moved funds sweep proof if it's `Live`
+			//  or `MovingFunds`. If the state is different skip it.
+			continue
+		}
+
+		if wallet.PendingMovedFundsSweepRequestsCount == 0 {
+			// If the wallet does not have any pending moved funds sweep
+			// requests skip it.
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply pkg/maintainer/spv/moved_funds_sweep.go:202]: Done in 76f0f43d6fac9802089235431938dfa5d3902016.

File: pkg/maintainer/spv/moved_funds_sweep.go [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+	for _, walletPublicKeyHash := range walletPublicKeyHashes {
+		wallet, err := spvChain.GetWallet(walletPublicKeyHash)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get wallet: [%v]", err)
+		}
+
+		if wallet.State != tbtc.StateLive &&
+			wallet.State != tbtc.StateMovingFunds {
+			// The wallet can only submit moved funds sweep proof if it's `Live`
+			//  or `MovingFunds`. If the state is different skip it.
```
[the remark this answers is not in the document]
Comment (tomaszslabon) [reply pkg/maintainer/spv/moved_funds_sweep.go:190]: Done in 76f0f43d6fac9802089235431938dfa5d3902016.

File: pkg/maintainer/spv/moving_funds_test.go (hunk 2 above)
Comment (tomaszslabon) [reply pkg/maintainer/spv/moving_funds_test.go:41]: This change was intentional.
I've noticed that the URLs in the file ended with `/` so I added it here.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**